### ISSUE TITLE
7zip is unable to locate archive + Playnite crashes after game is installed

### ISSUE THREAD

**Issue by wbstkr** [azuravian/playnite-LocalLibrary#4]
hello, this is probably my second or third issue ive ever made on github, so if i am not following normal etiquette please let me know as i am eager to learn.

OS: Windows 11
Playnite version: 10.34
Library version: 1.0.1

7zip is unable to locate archive:
1. add local game
2. under installation, add rom with name "Install" and path to a .rar file containing local game
3. under the addon settings for this library i set the 7zip path to the path to my 7z.exe
4. go back to local game and click on install
5. set installation path to folder i want the archive to extract to
6. extraction does not happen
7. im confused so i change the 7zip path to the path to my 7zG.exe
8. repeat 4 and 5
9. 7zip is telling me that it is unable to locate archive

Playnite crashes after game is installed
10. i change the 7zip path to the path to my 7zFM.exe
11. repeat 4 and 5
12. this time the file manager opens up and I extract the archive myself
13. after closing the the file manager, playnite prompts me to select the game's exe
14. after selecting the game's exe, playnite freezes for a few seconds and crashes
15. reopen playnite
16. game has been installed

as you can see, the game does get installed. however, i would assume playnite crashing is not an intentional thing, and i think its a little annoying that i have to extract the archives myself, because at that point i could just be doing this process without this plugin entirely. if there's something that im doing to cause this issue to occur let me know, and if you need more details let me know as well.

**Comment by azuravian** [comment]
Yep, found the issue here.  I'll have a new release out shortly.  Your first executable is the correct one to use, there was just an issue with the paths that caused trouble.  I've fixed this issue, but am also finalizing a new function that will detect games in your install folder and create the game entry with the correct ROM or Game Action applied.  As soon as that's done, I'll update which will fix this as well.

**Comment by wbstkr** [comment]
oh sweet! thats great, i cant wait

**Comment by azuravian** [comment]
This is resolved in [v1.5.0](https://github.com/azuravian/playnite-LocalLibrary/releases/tag/1.5.0)

**Comment by wbstkr** [comment]
Thanks! the 7zip issue has been resolved. However, Playnite still crashes when I select the .exe for the game after the archive has been extracted.

**Comment by azuravian** [comment]
Hmmm, that definitely should not happen.  Does it give an error first?

**Comment by wbstkr** [comment]
Nope, i dont get an error. Playnite just immediately crashes and I am forced to reopen it manually. Again, when i reopen it, the game seems to have installed properly.

**Comment by azuravian** [comment]
Do you have any games with installers?  I want to see if the same things happen after that or if it's only after a 7zip extract.

**Comment by wbstkr** [comment]
I just tried a game that had an installer. It also crashes Playnite. I am attaching the extensions.log here (from before I reopened Playnite after it crashed). Reopening Playnite after it crashes and going to the game I tried to install shows that it was installed correctly, with the action being set up correctly.
[extensions.log](https://github.com/user-attachments/files/18046021/extensions.log)

**Comment by azuravian** [comment]
Sorry, I haven't been online for a few days.  Just looked through this and there's nothing in the extensions.log.  Run the process one more time to get a crash and send me the Playnite.log from the same folder.

**Comment by wbstkr** [comment]
no problem dude, heres the log:

[playnite.log](https://github.com/user-attachments/files/18133937/playnite.log)

**Comment by azuravian** [comment]
Closing issue - no response
